header-validator/src/validate-json.ts

```diff
@@ -405,14 +456,20 @@ function legacyDuration(ctx: Context, j: Json): Maybe<number | bigint> {
 function collection<C extends context.PathComponent>(
   ctx: Context,
   js: Iterable<[C, Json]>,
-  f: CtxFunc<[C, Json], Maybe<unknown>>
+  f: CtxFunc<[C, Json], Maybe<unknown>>,
+  keepGoing: boolean = true
 ): boolean {
   let ok = true
   for (const [c, j] of js) {
     ctx.scope(c, () => {
       let itemOk = false
       f(ctx, [c, j]).peek(() => (itemOk = true))
-      ok = ok && itemOk
+      if (!itemOk) {
+        if (!keepGoing) {
```

Comment:
Would this only resolve the current callback without actually exiting the top function ?